cukas.js

```diff
@@ -106,6 +107,22 @@ class CukasGame {
             this.players[defencePlayerId].hand.push(this.deck.getRandomSet(Math.abs(this.players[attackPlayerId].hand.length - 6))); //make sure defence has 6 cards
         }
         //console.log(this.players);
+
+    }
+    compareCards(A,B){//1 - a>b -1 - a<b 0 - a=b
+      if(A.type==Card.TYPE_JOKER)return 1;
+      if(B.type==Card.TYPE_JOKER)return -1;
+      let a_trump=A.suit==trump;
+      let b_trump=B.suit==trump;
+      if(A.suit==B.suit || a_trump || b_trump){
+        let pow=A.rank;
+        if(a_trump)pow+=20;
+        pow-=B.rank;
+        if(b_trump)pow-=20;
+        return Math.sign(pow);
+      }
+      console.warn("Warning: these are not comparable");
```

Comment:
Kādos apstākļos šī rindiņa tiek sasniegta?

Reply:
Tad kad if-i neizpildaas.

Reply:
Ok, tik daudz es saprotu. Jautājums ir, kāpēc ifiem nebūtu jāizpildas. Turklāt, es skatos, ka kods referencē `trump` bez `this`. Tu savu kodu vispār testē, pirms uztaisi kārtējo commitu?

Reply:
Turklāt, ja `a.suit == b.suit`, tad kāda tur nozīme, vai a vai  b masts ir trump masts? Arī, ir neefektīvi if steitmentā vispirms notestēt (x || y), un tad body atsevišķi testēt (x) un tad (y). Šo atsevišķo testēšanu jāiznes galvenajā if/else-if steitmentu ķēdē.

Reply:
> 
> 
> Ok, tik daudz es saprotu. Jautājums ir, kāpēc ifiem nebūtu jāizpildas. Turklāt, es skatos, ka kods referencē `trump` bez `this`. Tu savu kodu vispār testē, pirms uztaisi kārtējo commitu?

oh, aizmirsu ielikts this, gruuti testeet CukasGame kad salabotaa versija veel name mergota.



> 
> 
> Turklāt, ja `a.suit == b.suit`, tad kāda tur nozīme, vai a vai b masts ir trump masts? Arī, ir neefektīvi if steitmentā vispirms notestēt (x || y), un tad body atsevišķi testēt (x) un tad (y). Šo atsevišķo testēšanu jāiznes galvenajā if/else-if steitmentu ķēdē.

so, es jau salaboju??
`compareCards(a,b){//1 - a>b -1 - a<b 0 - a=b
      if(a.type==Card.TYPE_JOKER && b.type!=Card.TYPE_JOKER)return 1;
      if(a.type==Card.TYPE_JOKER && b.type==Card.TYPE_JOKER)return 0;
      if(b.type==Card.TYPE_JOKER)return -1;
      const a_trump=a.suit==this.trump;
      const b_trump=b.suit==this.trump;
      if(a.suit==b.suit){
        return Math.sign(a.rank-b.rank);
      } else if(a_trump || b_trump) {
          if(a_trump)return 1;
          if(b_trump)return -1;
      } else {
          console.warn("Warning: these are not comparable");
          return false;
      }
    }`

Reply:
```js
...
} else if(a_trump) {
  return 1;
} else if(b_trump) {
  return -1;
} else {
...
```

Reply:
:ok_hand: